app/routes/_index.tsx

```diff
@@ -2,17 +2,34 @@ import type { LoaderArgs, LoaderFunction } from "@remix-run/node";
 import { redirect } from "@remix-run/node";
 
 import { getAuthSession } from "~/modules/auth";
+import { getAuthResponseByAccessToken } from "~/modules/auth/service.server";
 import { getUserByEmail } from "~/modules/user";
 
 export const loader: LoaderFunction = async ({ request }: LoaderArgs) => {
   const authSession = await getAuthSession(request);
 
-  const user = authSession
-    ? await getUserByEmail(authSession?.email)
-    : undefined;
+  // If there's no authSession, user isn't logged in. Redirect to login.
+  if (!authSession) {
+    return redirect("login");
+  }
+
+  const { data, error } = await getAuthResponseByAccessToken(
```

Comment:
I am a bit confused on why we need to do and handle this. Can we not handle this all on the onboarding page directly. Also I am a bit confused why in this case we need to find the auth using token, rather than just get it from the user. Can we not get this information from the user?